src/SOS/Strike/datatarget.cpp

```diff
@@ -129,6 +135,15 @@ DataTarget::ReadVirtual(
     {
         return E_UNEXPECTED;
     }
+#ifdef FEATURE_PAL
+    if (g_sos != nullptr)
+    {
+        if (IsMetadataMemory(address, request))
+        {
```

Comment:
This may also be a perf issue. If you had an app with many modules (asp.net loads a lot, more than 100?) and you ran an algorithm that is memory intensive (dumpheap, gcroot, stacktracing?) we will do work that is num_modules * num_reads. Its probably worth a quick manual test, but as long as nothing is unbearably slow I'm not worried.

Reply:
Building the metadata region list only happens once.  Are you worried about the cost of the std::set.find call?

I'll still do some manual perf testing.

Reply:
Yeah I was worried about the search... although looking closer and seeing that you have a O(log(N)) search rather than O(N) search reduced my concern considerably : )